src/Aspire.Hosting.Azure.EventHubs/AzureEventHubsExtensions.cs

```diff
@@ -99,4 +101,102 @@ public static IResourceBuilder<AzureEventHubsResource> AddEventHub(this IResourc
+    {
+        if (builder.ApplicationBuilder.ExecutionContext.IsPublishMode)
+        {
+            return builder;
+        }
+
+        builder.ApplicationBuilder.Services.TryAddLifecycleHook<AzureEventHubsEmulatorConfigWriterHook>();
+
+        // Add emulator container
+
+        builder
+            .WithEndpoint(name: "emulator", targetPort: 5672)
+            .WithAnnotation(new ContainerImageAnnotation
+            {
+                Registry = EventHubsEmulatorContainerImageTags.Registry,
+                Image = EventHubsEmulatorContainerImageTags.Image,
+                Tag = EventHubsEmulatorContainerImageTags.Tag
+            })
+            .WithAnnotation(new ContainerMountAnnotation(
+                Path.GetTempFileName(),
+                AzureEventHubsEmulatorResource.EmulatorConfigJsonPath,
+                ContainerMountType.BindMount,
+                isReadOnly: false));
+
+        // Create a separate storage emulator for the Event Hub one
+        var storageResource = builder.ApplicationBuilder
+                .AddAzureStorage($"{builder.Resource.Name}-storage")
+                .RunAsEmulator();
+
+        var storage = storageResource.Resource;
+
+        builder.WithAnnotation(new EnvironmentCallbackAnnotation((EnvironmentCallbackContext context) =>
+        {
+            var blobEndpoint = storage.GetEndpoint("blob");
+            var tableEndpoint = storage.GetEndpoint("table");
+
+            context.EnvironmentVariables.Add("ACCEPT_EULA", "Y");
+            context.EnvironmentVariables.Add("BLOB_SERVER", $"{blobEndpoint.ContainerHost}:{blobEndpoint.Port}");
+            context.EnvironmentVariables.Add("METADATA_SERVER", $"{tableEndpoint.ContainerHost}:{tableEndpoint.Port}");
+        }));
+
+        if (configureContainer != null)
+        {
+            var surrogate = new AzureEventHubsEmulatorResource(builder.Resource);
+            var surrogateBuilder = builder.ApplicationBuilder.CreateResourceBuilder(surrogate);
+            configureContainer(surrogateBuilder);
+        }
+
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a bind mount for the data folder to an Azure Event Hubs emulator resource.
+    /// </summary>
+    /// <param name="builder">The builder for the <see cref="AzureEventHubsEmulatorResource"/>.</param>
+    /// <param name="path">Relative path to the AppHost where emulator storage is persisted between runs. Defaults to the path '.eventhubs/{builder.Resource.Name}'</param>
+    /// <param name="isReadOnly">A flag that indicates if this is a read-only mount.</param>
+    /// <returns>A builder for the <see cref="AzureEventHubsEmulatorResource"/>.</returns>
+    public static IResourceBuilder<AzureEventHubsEmulatorResource> WithDataBindMount(this IResourceBuilder<AzureEventHubsEmulatorResource> builder, string? path = null, bool isReadOnly = false)
```

Comment:
Why don't we have a `WithDataVolume`?

Reply:
Also - what does a data folder do for Azure EventHubs? Doesn't it store all its necessary data in the Storage blobs we gave it?

Reply:
> Also - what does a data folder do for Azure EventHubs? Doesn't it store all its necessary data in the Storage blobs we gave it?

The EH uses the storage blobs for managing partition leases only, not for general data (messages.) The EH client uses a storage account to hold checkpointing info.

Reply:
@eerhardt The Azure Storage emulator is using `/data` for its data folder (at least that what our storage component assume, so I'll trust it). I assume this is correct here to do the same since we use the azure emulator too.

But I agree we should do the same with a volume, I will create the PR.

(NB: Was catching up on inclusive code reviews I missed and heard your comment about comments on closed PRs, then remembered this comment 🙇)